ui/src/views/repositories/modals/add-repository-modal/add-repository-from-csv-modal.tsx

```diff
@@ -1,16 +1,36 @@
 import { Label, Textarea } from '@mergestat/blocks'
-import React from 'react'
+import React, { ChangeEvent, useCallback, useEffect } from 'react'
+import { useRepositoriesSetState } from 'src/state/contexts'
+import { TEST_IDS } from 'src/utils/constants'
 
 export const AddRepositoryFromCSVModal: React.FC = () => {
-  const onChange = React.useCallback(() => null, [])
+  const { setReposToAdd } = useRepositoriesSetState()
+
+  const onChange = useCallback((e: ChangeEvent<HTMLTextAreaElement>) => {
+    const lines = e.target.value.split(/\r?\n/)
+    const matrix = lines.map(line => line.split(',').map(ele => ele.trim()))
+
+    let newRepos: string[] = []
+    matrix.forEach(line => {
+      if (line[0] !== '' && !newRepos.includes(line[0])) {
+        newRepos = [...newRepos, line[0]]
+      }
+    })
+    setReposToAdd(newRepos)
+  }, [setReposToAdd])
```

Comment:
is there a reason this needs to happen in `onChange` (vs when the Add button is clicked)? It seems possibly a bit expensive - I noticed on my machine it was a bit slow typing in the textarea - not sure if you see the same thing?

Reply:
I left this cuz it suggested me that, but now that you said it, confirm what I noticed as well, so.. I will remove it instead. let me take care of it then